R/profile_wrapper.R

```diff
@@ -56,21 +55,31 @@ profile_wrapper <- function(mydir, model_settings) {
     ), file = "run_diag_warning.txt")
     message(paste0("Running profile for ", para, "."))
 
-    # Copy the control file to run from the copy
-    if (!file.exists(file.path(profile_dir, "control.ss_new"))) {
-      orig_dir <- getwd()
-      setwd(profile_dir)
-      r4ss::run(
-        dir = profile_dir,
-        exe = model_settings$exe,
-        extras = model_settings$extras
-      )
-      setwd(orig_dir)
+    # check for whether oldctlfile exists
+    if (!file.exists(file.path(profile_dir, model_settings$oldctlfile))) {
+      # if the oldctlfile is control.ss_new, and doesn't exist,
+      # run the model to create it
+      if (model_settings$oldctlfile == "control.ss_new") {
+        if (model_settings$verbose) {
+          "running model to get control.ss_new file"
+        }
+        orig_dir <- getwd()
```

Comment:
It appears as though `r4ss::run()` will take care of setting the working directory and getting you back out of it so I propose that we remove lines 66, 67, and 73. The benefit of using the functionality within `r4ss::run()` is that you would no longer be doing things twice, i.e., quicker code, and {r4ss} uses `on.exit()` to ensure that users get back to where they were if the function fails for some reason.